src/rgw/rgw_admin.cc

```diff
@@ -10591,7 +10604,10 @@ int main(int argc, const char **argv)
       cerr << "ERROR: topic name was not provided (via --topic)" << std::endl;
       return EINVAL;
     }
-
+    if (!driver->is_meta_master()) {
```

Comment:
i think that this should be done after the persistent topic removal.
the queue is duplicated per zone.

Reply:
> i think that this should be done after the persistent topic removal.
> the queue is duplicated per zone.

Check the do-remove and do-add function inside the metadata handler. Svc-topic-rados.cc. We add/remove the persistent queue for secondary